client/src/app/pages/controls/stakeholders/stakeholders.tsx

```diff
@@ -61,8 +61,13 @@ export const Stakeholders: React.FC = () => {
   const [stakeholderIdToDelete, setStakeholderIdToDelete] =
     React.useState<number>();
 
-  const [isNewModalOpen, setIsNewModalOpen] = React.useState(false);
-  const [rowToUpdate, setRowToUpdate] = React.useState<Stakeholder>();
+  const [createUpdateModalState, setCreateUpdateModalState] = React.useState<
+    "create" | Stakeholder | null
+  >(null);
+  const isCreateUpdateModalOpen = createUpdateModalState !== null;
+  const stakeholderToUpdate =
+    createUpdateModalState !== "create" ? createUpdateModalState : null;
+
```

Comment:
We should move the modal refactor work to a new PR and address it systematically if we want to do it.

Reply:
The change with the Id, triggered a cascade of changes.
This is now aligned with our latest practice.
Sorry for the big PR :(